js/app/location.js

```diff
@@ -0,0 +1,142 @@
+define([
+    '../utility/check-data.js',
+],
+(CheckData) => {
+    var Location = function(parent, data) {
+        var myself = self;
```

Comment:
Why is this needed? Can't you just use 'self'?

Reply:
I added myself everywhere. so it all looks like the same class pattern but it is not used much.